apps/system/js/pin_page_system_dialog.js

```diff
@@ -12,12 +12,14 @@
     this.instanceID = 'pin-page-dialog';
     this.controller = controller || {};
     this.options = {};
-    this.render();
     LazyLoader.load('shared/elements/gaia-component/gaia-component.js')
       .then(() => {
         LazyLoader.load('shared/elements/gaia-site-icon/script.js');
+        LazyLoader.load('shared/elements/gaia-button/gaia-button.js');
+        LazyLoader.load('/shared/elements/gaia-dialog/gaia-dialog.js');
         LazyLoader.load('shared/elements/gaia-toast/gaia-toast.js')
           .then(() => {
+            this.render();
```

Comment:
This looks racy to me. We aren't waiting for gaia-button and gaia-dialog to load before rendering the entire dialog. I think we to wait for those two components to load (perhaps using Promise.all) before calling render.